src/Stratis.Features.FederatedPeg/TargetChain/WithdrawalTransactionBuilder.cs

```diff
@@ -50,20 +50,11 @@ public Transaction BuildWithdrawalTransaction(uint256 depositId, uint blockTime,
                     IgnoreVerify = true,
                     WalletPassword = walletPassword,
                     Sign = sign,
+                    Time = this.network.Consensus.IsProofOfStake ? blockTime : (uint?) null
                 };
 
-                Transaction transaction = this.federationWalletTransactionHandler.BuildTransaction(multiSigContext);
-
                 // Build the transaction.
-                if (this.network.Consensus.IsProofOfStake)
-                {
-                    transaction.Time = blockTime;
-
-                    if (sign)
-                    {
-                        transaction = multiSigContext.TransactionBuilder.SignTransaction(transaction);
```

Comment:
Hmm, this has disappeared, is it a change in functionality?

Reply:
No change in functionality. The above line `federationWalletTransactionHandler.BuildTransaction(multiSigContext)` is calling `TransactionBuilder.BuildTransaction(context.Sign)` which then calls `transactionBuilder.SignTransaction` anyway. So previously, the transaction was being build and signed, then having a timestamp added and signed again.

Now the timestamp is added to the builder and it's only built and signed once.